### ISSUE TITLE
AllOptions.yaml should be expanded with a couple missing options, and should provde a far througher documentation of all the options for md-flexible

### ISSUE THREAD

**Issue by SamNewcome** [AutoPas/AutoPas#706]
**Describe the solution you'd like**

- Every option should be included
- Every option should include all possible choices
- Some, less clear, options should have comments explaining them, which somebody new to AutoPas/md-flexible could understand.

**Additional context**
It would be better to handle this after MultiCenteredMolecules has been merged to master, as this branch changes how some options work, even for single-site simulations.

**Comment by FG-TUM** [comment]
> Some, less clear, options should have comments explaining them, which somebody new to AutoPas/md-flexible could understand

Not sure I agree, as this would make it necessary to keep this in sync with the parameter documentation in `MDFlexConfig`. So either we only write in `AllOptions.yaml` something like "for further information see `MDFlexConfig` or we generate AllOptions similar to the zsh completions. This could even be triggered via CMake whenever one builds md-flex.

**Comment by SamNewcome** [comment]
Yes, I see this point. But in the documentation, we point to `AllOptions.yaml` as the reference. And ideally, the user shouldn't have to scroll through `MDFlexConfig` to get information about what each option does (and the doxygen isn't much nicer). I agree that repeating the explanation in both `AllOptions.yaml` and `MDFlexConfig` is not a great idea, and I like the notion of something auto-generated. 

I would suggest, for now, a comment pointing `MDFlexConfig`, but I think a nicer solution is desirable in the long run. Potentially, when the documentation is built, it generates a page using the descriptions in `MDFlexConfig` + with an example like in, or directly from, `AllOptions.yaml` (exactly how this is done can be tbd).

**Comment by dmartin13** [comment]
I added a small comment to AllOptions.yaml in https://github.com/AutoPas/AutoPas/pull/752 that points to the description of parameters. In https://github.com/AutoPas/AutoPas/pull/732 I have already added all missing options. So maybe we can close this now.

**Comment by SamNewcome** [comment]
Yes, I agree!
